features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/BotDetection/EmailConfig/AddEmail.jsx

```diff
@@ -0,0 +1,169 @@
+ */
+
+import React, { useReducer } from 'react';
+import API from 'AppData/api';
+import PropTypes from 'prop-types';
+import Joi from '@hapi/joi';
+import TextField from '@material-ui/core/TextField';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import { FormattedMessage } from 'react-intl';
+import { makeStyles } from '@material-ui/core/styles';
+import FormDialogBase from 'AppComponents/AdminPages/Addons/FormDialogBase';
+import Alert from 'AppComponents/Shared/Alert';
+
+const useStyles = makeStyles((theme) => ({
+    error: {
+        color: theme.palette.error.dark,
+    },
+}));
+
+/**
+ * Reducer
+ * @param {JSON} state State
+ * @returns {Promise}.
+ */
+function reducer(state, { field, value }) {
+    return {
+        ...state,
+        [field]: value,
+    };
+}
+
+/**
+ * Render a pop-up dialog to add an email address
+ * @param {JSON} props .
+ * @returns {JSX}.
+ */
+function AddEmail(props) {
+    const {
+        updateList, icon, triggerButtonText, title, emailList,
+    } = props;
+    const classes = useStyles();
+    const initialState = {
+        email: '',
+
+    };
+
+    const [state, dispatch] = useReducer(reducer, initialState);
+    const { email } = state;
+
+    const onChange = (e) => {
+        dispatch({ field: e.target.name, value: e.target.value });
+    };
+
+    const validateEmail = (value) => {
+        const schema = Joi.string().email().empty();
+        const validationError = schema.validate(value).error;
+
+        if (validationError) {
+            const errorType = validationError.details[0].type;
+            if (errorType === 'any.empty') {
+                return 'Email is empty';
+            }
+            if (errorType === 'string.email') {
+                return 'Invalid Email';
+            }
+        } else {
+            const existingSameEmails = emailList.filter((obj) => obj.email === email);
+            if (existingSameEmails.length > 0) {
+                return 'Same email exists';
+            } else {
+                return false;
+            }
+        }
+        return false;
+    };
+
+    const formSaveCallback = () => {
+        const validationErrors = validateEmail(email);
+        if (validationErrors) {
+            Alert.error(validationErrors);
+            return false;
+        }
+
+        const restApi = new API();
+        const promiseAPICall = restApi.addBotDetectionNotifyingEmail(email);
+        return promiseAPICall.then(() => {
+            return (
+                <FormattedMessage
+                    id='AdminPages.BotDetection.Add.form.add.successful'
+                    defaultMessage='Email added successfully'
+                />
+            );
+        })
+            .catch((error) => {
+                const { response } = error;
+                if (response.body) {
+                    throw response.body.description;
+                }
+            })
+            .finally(() => {
+                dispatch('', '');
```

Comment:
Why we dispatch an empty event and empty value here ???
```
dispatch('', '');
```

to clear the things up ? if so why ?

Reply:
This is an unnecessary line of code and removed with commit: https://github.com/wso2/carbon-apimgt/pull/8591/commits/814f59930b3d3e7e8507a02109f852c3906e686a